teku/src/main/java/tech/pegasys/teku/cli/slashingprotection/RepairCommand.java

```diff
@@ -0,0 +1,160 @@
+      arity = "0..1")
+  private boolean checkOnlyEnabled = false;
+
+  @CommandLine.Option(
+      names = {"--update-all-enabled"},
+      paramLabel = "<BOOLEAN>",
+      description = "All validator slashing protection records should be updated.",
+      arity = "0..1")
+  private boolean updateAllEnabled = false;
+
+  @CommandLine.Option(
+      names = {"--with-slot"},
+      paramLabel = "<INTEGER>",
+      description =
+          "Update slashing protection files to contain the specified slot as a minimum."
+              + "Should be a future slot, or after when the validators stopped performing duties.\n"
+              + "NOTE: This can be calculated for most networks, and is generally not required.",
+      arity = "1")
+  private UInt64 suppliedSlot;
+
+  private final WeakSubjectivityInitializer wsInitializer = new WeakSubjectivityInitializer();
+
+  @Override
+  public void run() {
+    final Path slashProtectionPath =
+        SlashingProtectionCommandUtils.getSlashingProtectionPath(dataOptions);
+    SlashingProtectionCommandUtils.verifySlashingProtectionPathExists(
+        SUB_COMMAND_LOG, slashProtectionPath);
+    final Spec spec = eth2NetworkOptions.getNetworkConfiguration().getSpec();
+
+    final Optional<AnchorPoint> initialAnchor =
+        wsInitializer.loadInitialAnchorPoint(
+            spec, eth2NetworkOptions.getNetworkConfiguration().getInitialState());
+
+    final UInt64 computedSlot = getComputedSlot(initialAnchor, spec);
+    final UInt64 computedEpoch =
+        spec.atSlot(computedSlot).miscHelpers().computeEpochAtSlot(computedSlot);
+    final SlashingProtectionRepairer repairer =
+        new SlashingProtectionRepairer(SUB_COMMAND_LOG, updateAllEnabled);
+
+    repairer.initialise(slashProtectionPath);
+
+    if (repairer.hasUpdates()) {
+      if (!checkOnlyEnabled) {
+        confirmOrExit();
+        repairer.updateRecords(computedSlot, computedEpoch);
+      } else {
+        SUB_COMMAND_LOG.display("Updates have been skipped as --check-only-enabled was set.");
+      }
+    }
+    SUB_COMMAND_LOG.display("done");
+  }
+
+  private UInt64 getComputedSlot(final Optional<AnchorPoint> initialAnchor, final Spec spec) {
+    final TimeProvider timeProvider = new SystemTimeProvider();
+    if (suppliedSlot != null) {
+      displaySlotUpdateMessage(suppliedSlot, spec, "WARNING: using a supplied slot");
+
+      return suppliedSlot;
+    } else if (initialAnchor.isPresent()) {
+      final UInt64 genesisTime = initialAnchor.get().getState().getGenesis_time();
+      final int secondsPerEpoch =
+          spec.getGenesisSpec().getSlotsPerEpoch()
+              * spec.getGenesisSpec().getConfig().getSecondsPerSlot();
+      final UInt64 oneEpochInFuture = timeProvider.getTimeInSeconds().plus(secondsPerEpoch);
+      final UInt64 computedSlot =
+          SlashingProtectionCommandUtils.getComputedSlot(genesisTime, oneEpochInFuture, spec);
+      displaySlotUpdateMessage(computedSlot, spec, "Computed slot");
```

Comment:
Would probably be good to include the network name here - ideally `"Computed slot for mainnet"` or whatever.  Might depend a bit what we can actually get from the cli options - unfortunately the `CONFIG_NAME` constant is being removed from specs.